### PR TITLE
Add test push-variables-add-relation

### PR DESCRIPTION
**Pull request by webrouse**
Part of: https://keboola.atlassian.net/browse/COM-1137

### CODE REVIEW

File: test/functional_test.go [hunk 1]
```diff
@@ -294,7 +293,7 @@ func AssertExpectations(
 		// Compare expected and actual state
 		testhelper.AssertWildcards(
 			t,
-			json.MustEncodeString(expectedSnapshot, true),
+			testhelper.ReplaceEnvsString(expectedSnapshot.Content, envProvider),
```
Comment (webrouse) [inline test/functional_test.go:296]: Aby sme mohli v subore nahradit ENVs, tak ho nedekodujeme, iba nahradime placeholders.

File: internal/pkg/fixtures/fixtures.go [hunk 2]
```diff
@@ -42,15 +42,15 @@ type Config struct {
 	ComponentId       string                 `json:"componentId" validate:"required"`
 	Name              string                 `json:"name" validate:"required"`
 	Description       string                 `json:"description"`
-	ChangeDescription string                 `json:"changeDescription"`
+	ChangeDescription string                 `json:"changeDescription,omitempty"`
```
Comment (webrouse) [inline internal/pkg/fixtures/fixtures.go:45]: Preto sa prazdna `changeDescription` neuklada.

File: test/push-delete-config-row-dev-force/expected-state.json [hunk 3]
```diff
@@ -11,7 +11,6 @@
           "componentId": "keboola.ex-db-mysql",
           "name": "with-rows",
           "description": "test fixture",
-          "changeDescription": "",
```
Comment (webrouse) [inline test/push-delete-config-row-dev-force/expected-state.json:14]: Fixnute v 4 testoch, v ostatnych prazdna `changeDescription` nebola.

File: internal/pkg/testproject/snapshot.go [hunk 4]
```diff
@@ -79,7 +82,10 @@ func (p *Project) snapshot(snapshot *fixtures.ProjectSnapshot, configs map[strin
 								c.ChangeDescription = normalizeChangeDesc(config.ChangeDescription)
 								c.Content = config.Content
 								branchWithConfigs.Configs = append(branchWithConfigs.Configs, c)
+
+								lock.Lock()
 								configs[config.Key().String()] = c
```
Comment (webrouse) [inline internal/pkg/testproject/snapshot.go:87]: Bezi viacero paralelnych requestov, pristup do `map` musi byt chraneny zamkom.

File: test/push-variables-add-relation/expected-stdout [hunk 5]
```diff
@@ -0,0 +1,3 @@
+Plan for "push" operation:
+  CH C my-branch/extractor/ex-generic-v2/empty | changed: relations
```
Comment (webrouse) [inline test/push-variables-add-relation/expected-stdout:2]: Pridany novy test `push-variables-add-relation`:
- config + `variables` config existuju aj v `local` aj v `remote` stave.
- v `remote` ale nie su configs prepojene
- pri `push` sa configs prepoja, prida sa `variables_id`.

File: test/push-variables-add-relation/initial-state.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
+  "allBranchesConfigs": [],
+  "branches": [
+    {
+      "branch": {
+        "name": "Main",
+        "isDefault": true
+      },
+      "configs": [
+        "empty",
```
Comment (webrouse) [inline test/push-variables-add-relation/initial-state.json:10]: `inital` state, 2 nezavisle configs.

File: test/push-variables-add-relation/expected-state.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            }
+          ]
+        },
+        {
+          "componentId": "ex-generic-v2",
+          "name": "empty",
+          "description": "test fixture",
+          "changeDescription": "Updated from #KeboolaCLI",
+          "configuration": {
+            "variables_id": "%%TEST_BRANCH_MAIN_CONFIG_VARIABLES_ID%%",
```
Comment (webrouse) [inline test/push-variables-add-relation/expected-state.json:44]: expected state, pridane `variables_id` a `variables_values_id`.

**Review by JakubMatejka — APPROVED** [review]

**Pull request merged**